packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/InAppPurchaseStoreKit2.swift

```diff
@@ -34,4 +34,127 @@ extension InAppPurchasePlugin: InAppPurchase2API {
       }
     }
   }
+
+  // Gets the appropriate product, then calls purchase on it.
+  // https://developer.apple.com/documentation/storekit/product/3791971-purchase
+  func purchase(
+    id: String, options: SK2ProductPurchaseOptionsMessage?,
+    completion: @escaping (Result<SK2ProductPurchaseResultMessage, Error>) -> Void
+  ) {
+    Task {
+      do {
+        let product = try await Product.products(for: [id]).first
+        guard let product = product else {
+          let error = PigeonError(
+            code: "storekit2_failed_to_fetch_product",
+            message: "Storekit has failed to fetch this product.",
+            details: "Storekit has failed to fetch this product.")
+          return completion(.failure(error))
+        }
+
+        let result = try await product.purchase(options: [])
+
+        switch result {
+        case .success(let verification):
+          switch verification {
+          case .verified(let transaction):
+            DispatchQueue.main.async {
```

Comment:
Nit: can use `Task { @MainActor in }` instead of GCD since we are not using GCD in `Task`